dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java

```diff
@@ -244,4 +297,633 @@ public final Response switchSite(
+                this.siteHelper.getSiteNoFrontEndRoles(user, hostId);
+
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+
+        this.siteHelper.unpublish(host, user, pageMode.respectAnonPerms);
+        return Response.ok(new ResponseEntityView(this.toView(host))).build();
+    }
+
+    /**
+     * Archive a host
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @PUT
+    @Path("/_archive/{hostId}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response archiveSite(@Context final HttpServletRequest httpServletRequest,
+                                @Context final HttpServletResponse httpServletResponse,
+                                @PathParam("hostId")  final String hostId) throws DotDataException, DotSecurityException, PortalException, SystemException {
+
+        final User user = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(httpServletRequest, httpServletResponse)
+                .requiredBackendUser(true)
+                .rejectWhenNoUser(true)
+                .requireLicense(true)
+                .init().getUser();
+
+        Logger.debug(this, ()-> "Archiving site: " + hostId);
+
+        final PageMode      pageMode      = PageMode.get(httpServletRequest);
+        final Host host = pageMode.respectAnonPerms? this.siteHelper.getSite(user, hostId):
+                this.siteHelper.getSiteNoFrontEndRoles(user, hostId);
+
+        if (null == host) {
+            throw new IllegalArgumentException("Site: " + hostId + " does not exists");
+        }
+
+        if(host.isDefault()) {
+
+            throw new DotStateException("the default site can't be archived");
+        }
+
+        this.archive(user, pageMode, host);
+        return Response.ok(new ResponseEntityView(this.toView(host))).build();
+    }
+
+    @WrapInTransaction
+    private Response archive(final User user, final PageMode pageMode,
+                         final Host host) throws DotDataException, DotSecurityException {
+
+        if(host.isLocked()) {
+
+            this.siteHelper.unlock(host, user, pageMode.respectAnonPerms);
+        }
+
+        this.siteHelper.archive(host, user, pageMode.respectAnonPerms);
+        return Response.ok(new ResponseEntityView(this.toView(host))).build();
+    }
+
+    /**
+     * Unarchive a host
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param hostId
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     */
+    @PUT
+    @Path("/_unarchive/{hostId}")
```

Comment:
I think it's better to have the hostId and then the action the exec. `/{hostId}/_unarchive`